dotnet-desktop-guide/net/winforms/controls-design/usercontrol-guidelines.md

```diff
@@ -0,0 +1,51 @@
+---
+title: User control guidelines
+description: This article offers practical advice on how to design your user control for Windows Forms.
+ms.date: 08/08/2023
+ms.topic: overview
+dev_langs:
+  - "csharp"
+  - "vb"
+f1_keywords: 
+  - "UserControl"
+helpviewer_keywords: 
+  - "events [Windows Forms], defining within Windows Forms custom controls"
+  - "custom controls [Windows Forms], events using code"
+  - "property [Windows Forms], property changed events"
+  - "property [Windows Forms], designer"
+---
+
+# Guidelines to designing user controls (Windows Forms .NET)
+
+This article provides guidelines for creating user controls. We recommend that you follow these guidelines to make sure that you design a user control that's consistent with other Windows Forms controls.
+
+[!INCLUDE [desktop guide under construction](../../includes/desktop-guide-preview-note.md)]
+
+## Defining events
+
+Events commonly communicate state change and alert you to how the user is interacting with a Windows Forms control. For more information about events and delegates, see [Handle and raise events](/dotnet/standard/events/index).
+
+When defining your own events, follow these suggestions:
+
+- Use the <xref:System.EventHandler> event delegate when you define an event that doesn't have any associated data. Use the <xref:System.EventHandler%601> event delegate when you do have associated data.
```

Comment:
Not sure I understand what associated data means here. Data, which requires modified/derived `EventArgs`?
In that case, we would need to define respective `EventHandler` which match the `EventArgs` type (I wasn't sure what the difference here was in the two links you provided, btw.)

One thing people could use is this (and maybe that was what the link led to?):
https://learn.microsoft.com/dotnet/api/system.eventhandler-1?view=net-7.0

But! I NEVER tested that with the OOP-Designer, to be honest, and we need to try that out. 😸